src/components/Explore/GroupBySelector.tsx

```diff
@@ -35,6 +38,38 @@ export function GroupBySelector({ options, radioAttributes, value, onChange, sho
   const { value: metric } = getMetricVariable(model).useState();
   const metricValue = metric as MetricFunction;
 
+  useEffect(() => {
+    const loadClickCounts = async () => {
+      try {
+        const storedCounts = await storage.getItem(GROUP_BY_CLICK_COUNTS_STORAGE_KEY);
+        if (storedCounts) {
+          const counts = JSON.parse(storedCounts);
+          setClickCounts(counts);
+        }
+      } catch (e) {
+        console.error('Failed to load group by click counts:', e);
+      }
+    };
+    
+    loadClickCounts();
+  }, []);
+
+  const trackOptionClick = async (option: string) => {
+    try {
+      const updatedCounts = { ...clickCounts };
+      updatedCounts[option] = (updatedCounts[option] || 0) + 1;
+      setClickCounts(updatedCounts);
+      await storage.setItem(GROUP_BY_CLICK_COUNTS_STORAGE_KEY, JSON.stringify(updatedCounts));
```

Comment:
The issue description says

> This would work per user and browser, without any configuration

but we save it to user storage in the backend. Is it what we want in this case or it could be an overkill for the storage?

Reply:
I think that is not a requirement that it's per browser, that was just a description based on the conversation we had at the time. Userstorage is the preferred Grafana way to store user preferences etc so not sure how you would measure if this is overkill.